### PR TITLE
extract schema datatype based on first row of the sql output.

### PR DESCRIPTION
**Pull request by tovganesh**
extract schema datatype based on first row of sql output. currently only string, number and boolean types are inferred. defaults to string.

### CODE REVIEW

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 1]
```diff
@@ -284,7 +284,7 @@ def getJobOutput(self, job):
 
         no_of_retries = 0
 
-        while len(res_lines) == 0:
+        while len(res_lines) <= 3:  # ensure that enough lines are present to start parsing
```
Comment (sujitkp-blr) [inline dbt/adapters/spark_cde/cdeapisession.py:287]: Suggest to make it like MIN_LINES_FOR_PARSING insead of 3 directly

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 2]
```diff
@@ -284,7 +284,7 @@ def getJobOutput(self, job):
 
         no_of_retries = 0
 
-        while len(res_lines) == 0:
+        while len(res_lines) <= 3:
```
[the remark this answers is not in the document]
Comment (tovganesh) [reply dbt/adapters/spark_cde/cdeapisession.py]: this has been updated with a comment

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 3]
```diff
@@ -334,12 +334,65 @@ def getJobOutput(self, job):
...
+        if (len(rows) > 0):
+            try:
+                schema, rows = self.guessDatatypes(schema, rows)
+            except Exception as e:
+                import traceback
+                print(traceback.format_exc())
+
+        return schema, rows
+
+    def guessDatatypes(self, schema, rows):
```
[the remark this answers is not in the document]
Comment (tovganesh) [reply dbt/adapters/spark_cde/cdeapisession.py]: updated.

File: dbt/adapters/spark_cde/cdeapisession.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/spark_cde/cdeapisession.py]: updated.

**Review by sujitkp-blr — APPROVED** [review]
ok for current data types , we need to check for more complex data types.

**Pull request merged**